std/algebra/emulated/sw_emulated/hints.go

```diff
@@ -70,42 +93,362 @@ func decomposeScalarG1Signs(mod *big.Int, inputs []*big.Int, outputs []*big.Int)
 	})
 }
 
-func decomposeScalarG1(mod *big.Int, inputs []*big.Int, outputs []*big.Int) error {
+// TODO @yelhousni: generalize for any supported curve.
+// as it currently works only for P-256, P-384 and STARK curve.
+func scalarMulG1Hint(_ *big.Int, inputs []*big.Int, outputs []*big.Int) error {
```

Comment:
Why the functions `scalarMulG1Hint` and `scalarMulGLVG1Hint` are separate, couldn't we use the same function?